### ISSUE TITLE
An issue while using test mode

### ISSUE THREAD

**Issue by omar-shaaban742001** [videturfortuna/vehicle_reid_itsc2023#3]
python teste.py --path_weights ./logs/VERIWILD/MBR_4G/0/

How should I use this command ?

I have tried to change this with my params but I got an error  "./logs/VERIWILD/MBR_4G/0/" this file not found 

Thank you for your time

**Comment by videturfortuna** [comment]
Hi,
If that folder does not exist perhaps you have not trained the model yet or your train save went to another folder name. The folder logs is only created after you train a model, and it is composed of 
/logs/DATASET_NAME/ARCHICTECTURE_NAME/EXPERIENCE_NUMBER/

If you alreadyu trained check if any folder exists inside logs it should contain the weights and config used to train.

If you did not train the model, to train the MBR_4G on VERIWILD you should first use:
To train:
```console
python main.py --model_arch MBR_4G --config ./config/config_BoT_VERIWILD.yaml
```
Only after training you may use the teste.py. You should also replace the path of the weights with yours which will be something like that "./logs/VERIWILD/MBR_4G/your_number/"

If you do not want to train the model you may download trained weights from my google drive and test, for example: 
python teste.py --path_weights YOUR_PATH/VeriWild/Hybrid_4B_new/

**Comment by omar-shaaban742001** [comment]
Hi , 
!python teste.py --path_weights /content/drive/MyDrive/Hybrid_4B_new/ 
I have tried this command and I have acsess to veriwild dataset and I want to use your waights to test this dataset but I got this error 
/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 20 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
Selected device: cuda
Using cache found in /root/.cache/torch/hub/XingangPan_IBN-Net_master
Using cache found in /root/.cache/torch/hub/XingangPan_IBN-Net_master
Query infer (%):   0% 0/24 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/content/vehicle_reid_itsc2023/teste.py", line 280, in <module>
    cmc, mAP = test_epoch(model, device, data_q, data_g, data['model_arch'], remove_junk=True, scaler=scaler)
  File "/content/vehicle_reid_itsc2023/teste.py", line 99, in test_epoch
    for image, q_id, cam_id, view_id  in tqdm(dataloader_q, desc='Query infer (%)', bar_format='{l_bar}{bar:20}{r_bar}'):
  File "/usr/local/lib/python3.10/dist-packages/tqdm/std.py", line 1182, in _iter_
    for obj in iterable:
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 630, in _next_
    data = self._next_data()
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 1345, in _next_data
    return self._process_data(data)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 1371, in _process_data
    data.reraise()
  File "/usr/local/lib/python3.10/dist-packages/torch/_utils.py", line 694, in reraise
    raise exception
RuntimeError: Caught RuntimeError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/worker.py", line 308, in _worker_loop
    data = fetcher.fetch(index)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/content/vehicle_reid_itsc2023/data/triplet_sampler.py", line 51, in _getitem_
    image = torchvision.io.read_image(img_name)
  File "/usr/local/lib/python3.10/dist-packages/torchvision/io/image.py", line 258, in read_image
    data = read_file(path)
  File "/usr/local/lib/python3.10/dist-packages/torchvision/io/image.py", line 52, in read_file
    data = torch.ops.image.read_file(path)
  File "/usr/local/lib/python3.10/dist-packages/torch/ops.py", line 692, in __call_
    return self._op(*args, **kwargs or {})
RuntimeError: [Errno 2] No such file or directory: '/home/eurico/VERI-Wild/images/20342/209530.jpg'

should I change any links from the teste.py

**Comment by videturfortuna** [comment]
On the config file located in /content/drive/MyDrive/Hybrid_4B_new/config.yaml you should change variable "ROOT_DIR" to your path of the images folder:
ROOT_DIR: YOUR_PATH/VERI-Wild/images/ 

If you want to avoid that warning you can also change variable "num_workers_teste" to 1 or 2.
num_workers_teste: 2


I believe you have already done this but still, on the teste.py file you should also change paths in lines 209-210 to your path of the query and gallery txt files:
'YOUR_PATH/train_test_split/test_3000_id_query.txt'
'YOUR_PATH/train_test_split/test_3000_id.txt'

I will change the repository this week so no paths have to be written inside the code, and write some try excepts with warnings so it becames easier to use.

**Comment by omar-shaaban742001** [comment]
Hi,
I have done all of these things but still I have the same error.
RuntimeError: [Errno 2] No such file or directory: '/home/eurico/VERI-Wild/images/20342/209530.jpg'

Sorry for wasting time.

**Comment by omar-shaaban742001** [comment]
Dear Eurico,

I hope this message finds you well. I wanted to take a moment to express my heartfelt gratitude for your remarkable paper and the invaluable code you've shared. Your work has been instrumental in guiding my project and has been a source of immense inspiration.

The code you provided has been incredibly helpful in advancing my understanding and implementation. It has allowed me to make significant progress in my endeavors, and for that, I am truly thankful. Your expertise and dedication are truly commendable, and I feel fortunate to have had the opportunity to leverage your insights.

I do have a query regarding the output generated from your code. I've successfully executed it, and while I'm able to view numerical results, I'm curious if there's a method or specific instructions to visualize the output as an image. Your guidance or suggestions on this matter would be immensely appreciated.

Once again, thank you for your outstanding contribution to the field. Your work continues to make a profound impact, and I'm deeply grateful for your generosity in sharing your expertise.

Looking forward to your guidance, and I remain appreciative of your support and contributions.

Warm regards,

Omar
